### PR TITLE
Task/wp 209 header, footer main menu

### PR DESCRIPTION
**Pull request by sophia-massie**
## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WG-209](https://tacc-main.atlassian.net/browse/WG-209)

## Summary of Changes: ##

- Adds image assets
- Creates header component to be used by Map and Main Menu page
- Updates project listing styles to make scroll table and have main menu children take up whole page

## Testing Steps: ##
1. Login to hazmapper and play with width/height. 

## UI Photos:
Screenshot 2024-12-03 at 11 09 12 AM

### CODE REVIEW

File: react/src/components/HazmapperHeader/HazmapperHeader.module.css [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.root {
```
Comment (nathanfranklin) [inline react/src/components/HazmapperHeader/HazmapperHeader.module.css:1]: can this have a fixed height of --hazmapper-top-navbar-height from here https://github.com/TACC-Cloud/hazmapper/blob/main/react/src/hazmapper.css#L16.  

also, there is placeholder in MapProject ([here](https://github.com/TACC-Cloud/hazmapper/blob/main/react/src/pages/MapProject/MapProject.tsx#L154) and the style [here](https://github.com/TACC-Cloud/hazmapper/blob/main/react/src/pages/MapProject/MapProject.tsx#L154)) that can not be removed and this used 💯

Comment (nathanfranklin) [reply react/src/components/HazmapperHeader/HazmapperHeader.module.css:1]: Related to the above, I think we should unify the names of this component and the CSS variable.  🤔  

what about:

`HeaderNavBar` and `hazmapper-header-navbar-height`

File: react/src/components/Projects/ProjectListing.tsx [hunk 2]
```diff
@@ -52,54 +52,79 @@ const ProjectListing: React.FC = () => {
...
-            data.map((proj) => (
-              <tr key={proj.id} onClick={() => navigateToProject(proj.uuid)}>
-                <td className={styles.mapColumn}>{proj.name}</td>
-                <td className={styles.projectColumn}>
-                  {proj.ds_project
-                    ? `${proj.ds_project?.value.projectId} |
+      <div className={styles.versionContainer}>
+        <img
+          src="./src/assets/Hazmapper-Stack@4x.png"
+          alt="Hazmapper Logo Version 2.17"
```
Comment (nathanfranklin) [inline react/src/components/Projects/ProjectListing.tsx:58]:
```suggestion
          alt="Hazmapper Logo"
```

File: react/src/pages/MainMenu/MainMenu.tsx [hunk 3]
```diff
@@ -16,39 +12,49 @@ const MainMenu = () => {
...
+    <div className={styles.root}>
+      <HazmapperHeader user={userData.username} />
+      <div className={styles.listingContainer}>
+        <ProjectListing />
+      </div>
+      <div className={styles.sponsorContainer}>
+        <a href="https://www.nsf.gov/">
+          <img
+            src="./src/assets/nsf.png"
+            alt="National Science Foundation webiste"
```
Comment (nathanfranklin) [inline react/src/pages/MainMenu/MainMenu.tsx:42]:
```suggestion
            alt="National Science Foundation website"
```

File: react/src/pages/MainMenu/MainMenu.tsx [hunk 4]
```diff
@@ -16,39 +12,49 @@ const MainMenu = () => {
...
+            width="60px"
+          />
+        </a>
+        <a href="https://www.designsafe-ci.org/">
+          <img
+            src="./src/assets/designsafe.svg"
+            alt="NHERI DesignSafe webiste"
```
Comment (nathanfranklin) [inline react/src/pages/MainMenu/MainMenu.tsx:49]:
```suggestion
            alt="NHERI DesignSafe website"
```

File: react/src/pages/MainMenu/MainMenu.tsx [hunk 5]
```diff
@@ -16,39 +12,49 @@ const MainMenu = () => {
...
+            width="200px"
+          />
+        </a>
+        <a href="https://www.designsafe-ci.org/about/">
+          <img src="./src/assets/nheri.png" alt="NHERI webiste" width="150px" />
```
Comment (nathanfranklin) [inline react/src/pages/MainMenu/MainMenu.tsx:54]:
```suggestion
          <img src="./src/assets/nheri.png" alt="NHERI website" width="150px" />
```

**Comment by sophia-massie** [comment]
Addressed requested changes to 
- error message not centered
- regression in delete modal
- factor out logo and user guide to main menu
- factor out user hook to nav bar only
- show login button when viewing public map, re-fetch user data on return 

Ready for another review

File: react/src/components/HeaderNavBar/HeaderNavBar.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  } = useAuthenticatedUser();
+
+  const handleLogin = () => {
+    const url = `/login?to=${encodeURIComponent(location.pathname)}`;
+    navigate(url);
+  };
+
+  useEffect(() => {
+    if (userData) refetch();
+  }, [userData, refetch]);
```
Comment (nathanfranklin) [inline react/src/components/HeaderNavBar/HeaderNavBar.tsx:26]: can the use of refetch be dropped?  i tested locally with viewing a public map and then logged in and all worked well without it.

**Review by nathanfranklin — APPROVED** [review]
LGTM. 👍 

I  just have one comment about the link to taggit user guide.

File: react/src/pages/MainMenu/MainMenu.tsx [hunk 7]
```diff
@@ -1,54 +1,61 @@
...
+        <ProjectListing />
 
-  return (
-    <>
-      <SectionHeader isNestedHeader>Main Menu</SectionHeader>
-      <div>
-        <InlineMessage type="info">
-          Welcome, {userData?.username || 'User'} <Icon name="user"></Icon>
-        </InlineMessage>
+        <Button
```
[the remark this answers is not in the document]
Comment (sophia-massie) [reply react/src/pages/MainMenu/MainMenu.tsx:20]: Added

**Pull request merged**